### PR TITLE
T5 prompt learning fixes

### PR DESCRIPTION
**Pull request by MaximumEntropy**
# What does this PR do ?

Fixes the following things for T5 prompt-learning:

1. T5 models with RPE.
2. Models trained with asymmetric enc-dec configurations where hidden size is present in cfg.encoder.hidden_size.
3. Aligns config variables names with the GPT prompt learning config.

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by vadam5 — CHANGES_REQUESTED** [review]
Just needs a few comments a couple places

File: examples/nlp/language_modeling/megatron_ckpt_to_nemo.py [hunk 1]
```diff
@@ -96,11 +97,25 @@ def convert(local_rank, rank, world_size, args):
...
+        args.pipeline_model_parallel_size > 1
+        and args.pipeline_model_parallel_split_rank is None
+        and args.model_type in ['t5', 'bart', 'nmt']
+    ):
+        if args.pipeline_model_parallel_split_rank is not None:
+            app_state.pipeline_model_parallel_split_rank = args.pipeline_model_parallel_split_rank
+        else:
+            if args.pipeline_model_parallel_size % 2 != 0:
+                raise ValueError(f"Pipeline model parallel size {args.pipeline_model_parallel_size} must be even if split rank is not specified.")
+            else:
```
[the remark this answers is not in the document]
Comment (MaximumEntropy) [reply examples/nlp/language_modeling/megatron_ckpt_to_nemo.py:111]: Just added a comment. You can still have a different number of enc/dec layers and set the middle rank as the split rank and things will work fine - its just not optimal for speed.

File: examples/nlp/language_modeling/megatron_ckpt_to_nemo.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ def get_args():
     parser.add_argument("--gpus_per_node", type=int, required=True, default=None)
     parser.add_argument("--tensor_model_parallel_size", type=int, required=True, default=None)
     parser.add_argument("--pipeline_model_parallel_size", type=int, required=True, default=None)
+    parser.add_argument("--pipeline_model_parallel_split_rank", type=int, required=False, default=None)
```
[the remark this answers is not in the document]
Comment (MaximumEntropy) [reply examples/nlp/language_modeling/megatron_ckpt_to_nemo.py:75]: Done!

**Review by vadam5 — APPROVED** [review]
LGTM!

**Pull request merged**